File: go/1179.go
package main

import "fmt"

func printAndClean(arr []int, t string) []int {
	for i, v := range arr {
		fmt.Printf("%s[%d] = %d\n", t, i, v)
	}

	return []int{}
}

func main() {
	var imp, par []int
	var n int

	for i := 0; i < 15; i++ {
		fmt.Scanf("%d", &n)

		if n%2 == 0 {
			par = append(par, n)
		} else {
			imp = append(imp, n)
		}

		if len(imp) == 5 {
			imp = printAndClean(imp, "impar")
		}

		if len(par) == 5 {
			par = printAndClean(par, "par")
		}
	}

	printAndClean(imp, "impar")
	printAndClean(par, "par")
}

File: lua/1179.lua
local function printAndClean(arr, t)
    for i, v in ipairs(arr) do
        io.write(string.format("%s[%d] = %d\n", t, i - 1, v))
    end
    return {}
end

local imp = {}
local par = {}
local n

for i = 1, 15 do
    n = io.read("*n")

    if n % 2 == 0 then
        table.insert(par, n)
    else
        table.insert(imp, n)
    end

    if #imp == 5 then
        imp = printAndClean(imp, "impar")
    end

    if #par == 5 then
        par = printAndClean(par, "par")
    end
end

printAndClean(imp, "impar")
printAndClean(par, "par")
